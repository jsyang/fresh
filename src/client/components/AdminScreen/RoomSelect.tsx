import {Component, h} from 'preact';
import store from '../../store';
import style from '../style';

const SelectField: any = style('select')({
    'font-size': '1em'
});

interface IRoomSelectProps {
    rooms?: { name: string }[];
    selected: string;
    onChange: Function;
}

export class RoomSelect extends Component<IRoomSelectProps> {
    render() {
        const {selected, onChange} = this.props;
        const rooms                = this.props.rooms || store.getState().rooms;

        return <SelectField onChange={onChange}>
            {rooms.map(({name}) => {
                const props: any = {value: name};
                if (selected === name) {
                    props.selected = 'selected';
                }

                return <option {...props}>{name}</option>;
            })}
        </SelectField>;
    }
}

export const getFormattedTaskValue = (value, keyName: string, onChange: Function) => {
    if (typeof value === 'boolean') {
        return value ? 'Y' : 'N';
    } else {
        if (keyName === 'room') {
            return <RoomSelect selected={value} onChange={onChange}/>;
        } else {
            return value;
        }
    }
};
