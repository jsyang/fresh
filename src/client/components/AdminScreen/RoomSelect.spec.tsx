import {h} from 'preact';
import {render} from 'preact-render-to-string';

import {RoomSelect} from './RoomSelect';

describe('components/RoomSelect', () => {
    it('should match snapshot', () => {
        expect(render(
            <RoomSelect
                rooms={[
                    {name: 'bed'},
                    {name: 'bath'},
                    {name: 'kitchen'}
                ]}
                selected="bed"
                onChange={new Function}
            />
        )).toMatchSnapshot();
    });
});
