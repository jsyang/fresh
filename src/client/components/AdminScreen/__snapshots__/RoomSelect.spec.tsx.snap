// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RoomSelect should match snapshot 1`] = `

<select class="p0">
  <option value="bed"
          selected="selected"
  >
    bed
  </option>
  <option value="bath">
    bath
  </option>
  <option value="kitchen">
    kitchen
  </option>
</select>

`;
